rand_core/src/lib.rs

```diff
@@ -185,22 +209,51 @@ pub trait SeedableRng<Seed>: Rng {
     fn from_seed(seed: Seed) -> Self;
 }
 
+/// Error type for random number generators.
+#[derive(Copy, PartialEq, Eq, Clone, Debug)]
+pub enum Error {
+    /// This RNG is permanently unavailable.
+    NotAvailable,
+    /// Needs (re)seeding. Depending on the RNG this can mean you have to wait
+    /// and retry later, or you have to (re)seed it.
+    EntropyLow,
```

Comment:
Not so keen on the doc or name for this one, and not sure if the two different things should be combined anyway — what's the advantage?

Possible doc: For PRNGs, this implies they should be (re)seeded. For external RNGs (e.g. `OsRng`) this implies you should wait and try again later.

Reply:
Ok, I will split them